### PR TITLE
Completed dealing with test requests

### PR DESCRIPTION
**Pull request by GeorgeF0**
- Completed dealing with test requests
  - engine does not keep sending test requests while the other end is not responding
  - session is terminated after two heartbeats' time of not receiving any messages
- Added more detail to the specification erroring

### CODE REVIEW

File: HotFix.Specification/Specification.cs [hunk 1]
```diff
@@ -64,13 +59,14 @@ public Specification Run()
             {
                 if (Exception != null)
                 {
+                    if (exception.GetType() == typeof(DelightfullySuccessfulException))
+                        throw new AssertFailedException("The expected exception type was not thrwon. All instructions successfully executed instead");
```
Comment (fedjabosnic) [inline HotFix.Specification/Specification.cs]: `throw new AssertFailedException($"The test expected an exception of type {Exception.FullName} but no exception was thrown");`

I think at this point we only care about the exception type/name so we want to print the *namespace and name*, aka `System.IO.IOException` etc

File: HotFix.Specification/Specification.cs [hunk 2]
```diff
@@ -64,13 +59,14 @@ public Specification Run()
...
+
                     if (Exception != exception.GetType())
-                        throw new AssertFailedException("The expected exception type was not thrown");
+                        throw new AssertFailedException($"The expected exception type was not thrown. The type '{exception.GetType()}' was thrown instead");
```
Comment (fedjabosnic) [inline HotFix.Specification/Specification.cs]: `throw new AssertFailedException($"The test expected an exception of type {Exception.FullName} but an exception of type {exception.GetType().FullName} was thrown instead");`

I think at this point we only care about the exception type/name so we want to print the *namespace and name*, aka `System.IO.IOException` etc

File: HotFix.Specification/Specification.cs (hunk 1 above)
Comment (GeorgeF0) [inline HotFix.Specification/Specification.cs]: Done

File: HotFix.Specification/Specification.cs (hunk 2 above)
Comment (GeorgeF0) [inline HotFix.Specification/Specification.cs]: Done

**Pull request merged**